Sources/Flatten.swift

```diff
@@ -389,10 +395,11 @@ extension SignalProtocol where Value: SignalProducerProtocol, Error == Value.Err
 						}
 					}
 				}
-				state.modify { $0.isStarting = false }
+
+				isStarting = false
```

Comment:
Is assigning to a `Bool` an atomic operation? This write could happen concurrently with the read above. I think it needs to at least be in its own `Atomic`.